cmd/detectExecuteScan.go

```diff
@@ -200,7 +203,12 @@ func runDetect(ctx context.Context, config detectExecuteScanOptions, utils detec
 	utils.SetDir(".")
 	utils.SetEnv(envs)
 
-	err = utils.RunShell("/bin/bash", script)
+	err = mapDetectError(utils.RunShell("/bin/bash", script), config, utils)
```

Comment:
do we intend to run the scans on the base artifact and the docker image as well ?. if the understand this will run the artifact scan and the `config.scanImages` will run the docker image scan. 

from a scan perspective do we want either you scan the artifact `or` the docker images  . Or do allow the flexibility to scan artifact `and` the docker image . with `and` i have some doubts about duplicates in the scans

Reply:
I think we do, yes. There might be relevant dependencies that don't make it into the image but should still be checked, like idk a Java dependency that ends up being packaged in some kind of fat jar and hence not be recognised as the dependency it is once it's put on the image.

Reply:
but there would be duplicate findings as well right ?, the docker scan will scan all layers in the image and when it scans  dependency manager layer, it will report all those vulnerabilities again ? .

in principle it must show all vulnerabilities in the same project version, but will be a good test in case of duplicates

Reply:
It's very likely worth checking if BD successfully detects duplicates. But I would hope BD will do the right thing or Synopsis agreeing that it does not and they should fix it.